### PR TITLE
internal/cryptutil: fixed potential panic on nil pubkey block

### PR DESCRIPTION
**Pull request by desimone**
It's an edge-case where if an empty public key was passed into `DecodePublicKey` a panic could be caused. 

**Checklist**:
- [x] unit tests added
- [x] ready for review

### CODE REVIEW

File: internal/cryptutil/marshal.go [hunk 1]
```diff
@@ -13,11 +13,9 @@ import (
 // DecodePublicKey decodes a PEM-encoded ECDSA public key.
 func DecodePublicKey(encodedKey []byte) (*ecdsa.PublicKey, error) {
 	block, _ := pem.Decode(encodedKey)
-	if block == nil || block.Type != "PUBLIC KEY" {
```
Comment (desimone) [inline internal/cryptutil/marshal.go]: To be clear `block.Type != "PUBLIC KEY" ` would panic if block ==nil here.
